### ISSUE TITLE
Please migrate to v2

### ISSUE THREAD

**Issue by dim** [drone/go-bitbucket#7]
BitBucket's v2 is now stable and is generally much, much better and simpler to use: 
https://confluence.atlassian.com/bitbucket/version-2-423626329.html

I did some quick checks and it still seems to work OK with OAuth1, so https://github.com/drone/drone/issues/1147 is a nice-to-have but not necessarily a pre-requisite.

**Comment by bradrydzewski** [comment]
any chance you know the answer to this:

> These tokens expire after just 1 hour before needing to be refreshed. We need to figure out how this impacts our overall system design. The reason is we make API calls from modules that have no access to the database and would be unable persist updated access_tokens when refreshed ... therefore I'm hoping that multiple access_tokens can exists at once (like github, google, etc allow) otherwise it is really going to complicate things.

before I invest more time in Bitbucket I really want to understand if their long term architecture is going to work well with Drone. When GitLab first implemented OAuth2, refreshing a token revoked all existing access tokens and it caused huge problems to the point where we had to revert back to basic auth. Having a single valid token in a parallel, distributed execution environment is problematic. GitLab eventually changed their default behavior.

**Comment by dim** [comment]
As said, this is independent of OAuth2 migration, as Bitbucket's V2 API accepts OAuth1 tokens (tried and tested). You could keep the https://github.com/drone/go-bitbucket/tree/master/oauth1 package and only update https://github.com/drone/go-bitbucket/tree/master/bitbucket

**Comment by bradrydzewski** [comment]
@dim the 0.4 version of Drone removed the abstraction tier that allowed oauth1 and oauth2 to run in parallel in anticipation of Bitbuckets oauth2 support. So unfortunately it isn't quite a drop-in replacement. We either need to re-implement an abstraction tier, or port to oauth2. I would prefer just implementing oauth2, but haven't had time to focus on it
